### ISSUE TITLE
Support Vertex Colors

### ISSUE THREAD

**Issue by ekt-** [nrgsille76/blender-io-addons#2]
Hi Sebastian,
first of all, thanks for the tool, I find it extremely useful. 
I think vertex colors are not supported. Attached you will find a file containing a mesh that has associated vertex colors. For the sake of this test, please focus on the object name 'factory-sky' (ignore Rectangle01 which you can delete).

![step1](https://github.com/nrgsille76/blender-io-addons/assets/4489148/80c6d4d8-8d28-41bd-9a3d-325d04771104)

After the import I see UV maps, but not vertex colors.
![step2-attributes](https://github.com/nrgsille76/blender-io-addons/assets/4489148/a4554d69-b5db-475c-bef2-cb24a0804136)
![step2-mesh](https://github.com/nrgsille76/blender-io-addons/assets/4489148/40de91ba-1445-4190-ab01-57d039bf6f90)

What is expected instead is this condition
![step3-expected-attributes](https://github.com/nrgsille76/blender-io-addons/assets/4489148/54240cfc-4a5a-44e1-bcf1-e64a95e1f1e5)
![step3-expected-colors](https://github.com/nrgsille76/blender-io-addons/assets/4489148/d896b4bc-d460-4086-aa73-83e664007c33)

python is definitely not my main language, but I can also try to check it out, if you give me some initial input. For example, I imagine that support for an object type should be added in object_filter = {'MATERIAL', 'UV', 'EMPTY'} and that create_shape( should somehow take care of it

attached repro file
[factory-sky.zip](https://github.com/user-attachments/files/16026117/factory-sky.zip)

**Comment by nrgsille76** [comment]
@ekt-  I already found those mappings along with the UVW mappings and Box mappings. The key for uvw maps is 0x1 and for Boxmap 0x2, the remaining key is 0x0 wich would be those vertex colors but I was not able to get floats or ints wich would make sense, it seems to be a byteflag array and without any information about the meaning of the flags I can not convert to blender API

**Comment by ekt-** [comment]
I understand. 
It doesn't add anything to the discussion, but one difference that comes to mind is that UVW maps are actually coordinates in texture space, while vertex colors should associate the vertex index with its color. What I mean is that they are actually two different things so they must be structured differently. And btw, vertex colors are 'archaic', from an era in which every byte counted and I wouldn't be surprised if there was some sort of 'palette indexing' involved.

Where can I find, if it exists, something that describes the max format? I'm having a bit of trouble reading Python but I could try parsing the file in C#/C++ with which I'm more comfortable and then report back to you if I find new info.

**Comment by nrgsille76** [comment]
@ekt- There is no information about the max format available in the internet. I used [this](https://github.com/decalage2/olefile) parser wich will deliver the file as directory containing the chunks. The relevant data is contained in the Scene chunk. There are some outcommented prints in my script wich can print the readable data like the directory or class names. The class you are looking for is called 'VertexPaint' wich inherits from UVW mapping because it always has the same vertex index like the UVW map.
I tried printing those data also in other files. In your file every vertex has the same value wich is (1.0, 1.0, 1.0) but I also got many files with nan values like this: 
![nan floats](https://github.com/nrgsille76/blender-io-addons/assets/61376632/9a2527de-e603-4c01-9e37-ce2e0a31b8b1)

**Comment by ekt-** [comment]
I just read about the structured storage format (a new exciting rabbit hole that I absolutely didn't need!). I can read the 'Scene' stream and now I'm looking at what you and [this other](https://github.com/jmplonka/Importer3D) do to read the 'Scene' stream chunks. all this in the unlikely event of finding something you already haven't spotted.
one thing that comes to mind for colors is that they could also be 24bit rgb color, and therefore represented by three bytes, not floats.

**Comment by ekt-** [comment]
an update.

TLDR; probably the problem is in the modifier stack

two premises

- for the moment I focused on 'editable poly'
- I wasn't clear which importer to use
\blender-io-addons\LTS\io_import_max.py
\blender-io-addons\io_autodesk_max\import_max.py
assuming the latest one was for blender4.2 I used the LTS one (I'm using 4.1). in any case the 'create_editable_poly' seems the same so it should be irrelevant.

the file I gave you initially is a bit strange, it could hide more than one problem. is composed of two 'editable poly':
- Rectangle01: editable poly, frozen, has 4 vertices, a (uv) map channel (with related texture) and an empty vc channel
- factory-sky: editable poly, 171 vertices and (only) the VertexChannel with data

Let's ignore Rectangle01 for now and focus on factory-sky

I instrumented the reading of the 0x128 section

```
elif (child.types == 0x0128):
   mesh.cords.append(calc_point_float(child.data))
```

I actually see all the points at [1,1,1] except the last four points
```
 float_point [1.000000,1.000000,1.000000]
 float_point [1.000000,1.000000,1.000000]
 float_point [0.733333,0.745098,0.643137]
 float_point [0.733333,0.745098,0.643137]
 float_point [0.827451,0.909804,0.874510]
 float_point [0.827451,0.909804,0.874510]
```

I'm sure that this file contains the vertex colors because I reinstalled 3dsmax (which is exactly what I didn't want to do and is the reason why I looked for an importer :_) and I can visually see that the vertex colors are there. So they're somewhere else.

The interesting thing is this:
factory-sky has a stack: UVW Mapping Clear/Vertex Paint/Editable Poly

if I collapse the stack, and it becomes just an 'Editable Poly' then in x0128 I see what many presumably are the rgb values ​​associated with each vertex

**Comment by ekt-** [comment]
Attached you will find two files, with the same 'Editable Poly' in the original version and with the collapsed modifier stack

![test-clean-wstack](https://github.com/nrgsille76/blender-io-addons/assets/4489148/2519529c-6de7-43c1-843f-8e68f9e2df38)
![test-clean-collapsed](https://github.com/nrgsille76/blender-io-addons/assets/4489148/13e19e8c-75e8-44ad-80c8-29d9b6a94787)

[test-clean-wstack.zip](https://github.com/user-attachments/files/16101138/test-clean-wstack.zip)
[test-clean-collapsed.zip](https://github.com/user-attachments/files/16101140/test-clean-collapsed.zip)

**Comment by nrgsille76** [comment]
@ekt-  the difference between LTS version and the newer versions are only in material setup since principled bsdf has been updated with parameter changed. The LTS version works for all versions the other version since blender 4.0
